### PR TITLE
Remove deprecated functionality from Google provider

### PR DESCRIPTION
**Pull request by moiseenkov**
Remove deprecated functionality that reached the end of life November 01, 2024.

### CODE REVIEW

File: providers/src/airflow/providers/google/CHANGELOG.rst [hunk 1]
```diff
@@ -27,6 +27,152 @@
...
 
+11.0.0
+......
+
+Breaking changes
+~~~~~~~~~~~~~~~~
+
+.. warning::
+  All deprecated classes, parameters and features have been removed from the Google provider package.
+  The following breaking changes were introduced:
```
Comment (potiuk) [inline providers/src/airflow/providers/google/CHANGELOG.rst:38]: :heart: the longlist of changes here :)

**Review by potiuk — APPROVED** [review]
I love those kinds of changes. @eladkal ?

**Comment by potiuk** [comment]
#protm

**Comment by potiuk** [comment]
I think you got big chances for PR of the month @moiseenkov :+1: 

image

File: providers/src/airflow/providers/google/CHANGELOG.rst (hunk 1 above)
Comment (moiseenkov) [reply providers/src/airflow/providers/google/CHANGELOG.rst:38]: Sorry, here are not all deprecations removed. Only those that reached their EOL.

Comment (moiseenkov) [reply providers/src/airflow/providers/google/CHANGELOG.rst:38]: I updated the changelog.

**Review by eladkal — APPROVED** [review]

**Review by kosteev — APPROVED** [review]

**Review by shahar1 — APPROVED** [review]
IMO - PROTM, less is definitely more in this case :)
Well done!

**Pull request merged**